contracts/swap/swap.go

```diff
@@ -99,6 +107,26 @@ func InstanceAt(address common.Address, backend Backend) (Contract, error) {
 	return c, err
 }
 
+func (s simpleContract) Withdraw(auth *bind.TransactOpts, backend Backend, amount *big.Int) (*types.Receipt, error) {
+	tx, err := s.instance.Withdraw(auth, amount)
+	if err != nil {
+		return nil, err
+	}
+	receipt, err := WaitFunc(auth, backend, tx)
+	return receipt, err
+}
+
+func (s simpleContract) Deposit(auth *bind.TransactOpts, backend Backend, amount *big.Int) (*types.Receipt, error) {
+	rawSimpleSwap := contract.SimpleSwapRaw{Contract: s.instance}
+	auth.Value = amount
```

Comment:
Yes, that is possible. I did it deliberately like this, as putting the amount inside auth might make it difficult to see what is actually happening. 

With hindsight, I am going to remove the `deposit` parameter, as it might happen that a developer sets `auth.Value` and `Deposit`, in which case it is not directly clear what happens/should happen.

Reply:
> With hindsight, I am going to remove the `deposit` parameter, as it might happen that a developer sets `auth.Value` and `Deposit`, in which case it is not directly clear what happens/should happen.

i agree with this change.

the only thing i would change now is to call this parameter `opts` instead of `auth` (is it somehow related to some kind of authorization that i am not seeing?)

Reply:
In `swap/contracts`, it is called `auth` everywhere. `auth` makes sense, as it also includes 		`opts := bind.NewKeyedTransactor(s.owner.privateKey)`

`opts` and `auth` are both no perfect names, but I would just leave it as is (preference for `auth`, as this is how it is called in the binaries generated by `abigen`)

Reply:
my problem with `auth` is that it implies _authorization_, and this deals with things such as _amount of wei to deposit_, which don't really have anything to do with this.

i understand `abigen` generates this, but is this dependent on the way it is written in the smart contract?

Reply:
In abigen, this variable is marked `auth` for a transaction that writes to the blockchain. It is called `opts` for a read-only operation.
In this case, `auth` does also imply authorization, as before we added the `bind.NewKeyedTransactor(s.owner.privateKey)` to the variable. 
If you want to change it, I propose that we do this in the renaming PR, as changing it in this PR will mean many more changes (we cannot just only change it here).

Reply:
sounds good :+1: